Devs-AA/Barefoot
# #168662119 Like/Unlike an accommodation facility

#### What does this PR do?
Enables users to like or unlike an accommodation facility

#### Description of Task to be completed?
- A requester can like accommodation he/she has visited
- A requester can unlike accommodation he or she has visited
- Only users who have visited the accommodation can like or unlike the facility

#### How should this be manually tested?
- git clone this repo using the command `git@github.com:Devs-AA/Barefoot.git`
  - cd to the project
  - checkout to the branch using the command `git checkout ft-like-unlike-accommodation-168662119`.
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below in the request body.
- `{email: requester2@gmail.com, password: Password1$}`
2. Copy the token in the reponse
3. Make a POST request to `http:localhost:<port>/api/v1/requests` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`
 4. Send a request object containing the following for a one-way trip:
         `{
           reason: 'Business',
           departmentId: 2,
           tripType: 'oneWay',
        }`

5. Make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below  in the request body.
 `{email: barefoot2@gmail.com, password: Password1$}`
6. Make a PATCH request to `http:localhost:<port>/api/v1/requests<requestId>` with the request body containing. Note that status can only be approved or rejected
          `{
            "status": "approved"
           }`
 7. Make a POST request to `http://localhost:<port>/api/v1/accommodations/<accommodationId>`, with the same authorization header used in creating request. The request body should comply with that in the image below
![Screenshot (103)](https://user-images.githubusercontent.com/46751363/69768451-e5f8df00-1180-11ea-83e9-d8efe5f3708f.png)

8. Make a PATCH request to `http://localhost:<port>/api/v1/accommodations/<accommodationId>`. With the request body as below. Note that `true` can be changed to `false`  for > unlike

![Screenshot (110)](https://user-images.githubusercontent.com/46751363/69835293-bdd0b500-1240-11ea-9435-75f5018064d5.png)

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168662119](https://www.pivotaltracker.com/story/show/168662119)

#### Screenshots (if appropriate)

![Screenshot (109)](https://user-images.githubusercontent.com/46751363/69835252-6599b300-1240-11ea-9c90-38fffe57a303.png)

![Screenshot (111)](https://user-images.githubusercontent.com/46751363/69835261-74806580-1240-11ea-90f4-9092529c4f6e.png)

![Screenshot (112)](https://user-images.githubusercontent.com/46751363/69835265-8235eb00-1240-11ea-96a9-796fb84bbfb1.png)


#### Questions
N/A
